nemo/collections/nlp/data/datasets/datasets_utils/data_preprocessing.py

```diff
@@ -314,19 +319,32 @@ def dataset_to_ids(dataset, tokenizer, cache_ids=False, add_bos_eos=True):
     return ids
 
 
-def calc_class_weights(label_freq):
+def get_freq_weights(label_freq):
     """
     Goal is to give more weight to the classes with less samples
-    so as to match the one with the higest frequency. We achieve this by
-    dividing the highest frequency by the freq of each label.
-    Example -
-    [12, 5, 3] -> [12/12, 12/5, 12/3] -> [1, 2.4, 4]
+    so as to match the ones with the higher frequencies. We achieve this by
+    dividing the total frequency by the freq of each label to calculate its weight.
+    """
 
-    Here label_freq is assumed to be sorted by the frequency. I.e.
-    label_freq[0] is the most frequent element.
+    total_size = 0
+    for lf in label_freq.values():
+        total_size += lf
+    weighted_slots = {label: (total_size / freq) for label, freq in label_freq.items()}
+    return weighted_slots
+
+
+def fill_class_weights(weights, max_id=-1):
```

Comment:
could you please add docstings to this one?

Reply:
Added. Also set weights for missing classes to one. It was zero before that. What do you think?

Reply:
I think it shouldn't matter since these labels are missing in the training data